### ISSUE TITLE
SearchBar is extremely laggy on physical device

### ISSUE THREAD

**Issue by codedbycurtis** [dotnet/maui#9242]
### Description

When entering text into the `SearchBar`, typing at a standard pace will cause some inputs to be missed. The only way to avoid this is by typing very slowly (e.g. one character per second). This issue does not appear on any Android emulators, but does on my physical device, and occurs in both debug and release mode. This leads me to believe that it may be device / operating system-specific.

I'm using a OnePlus 7 Pro running Oxygen OS 11.

I was not able to test any other physical devices.

### Steps to Reproduce

1. Create a new .NET Maui project.
2. Create a `ContentPage` with a `SearchBar` control.
3. Observe the lag when typing in the `SearchBar`.

### Version with bug

6.0.408 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

**Comment by codedbycurtis** [comment]
I have provided a repro [here](https://github.com/codedbycurtis/SearchBarRepro), however I don't think it will be of much help as it appears to work on a majority of devices.

Below is a video comparison between text input behaving correctly with the `Entry` control, and it behaving incorrectly with the `SearchBar` control.

Side note: I also noticed another strange issue when recording - the backspace key 'sticks' as well, and only eliminates one character at a time.

https://user-images.githubusercontent.com/69975259/183465191-a4926bfa-884f-45a5-ae8c-afd20fd0c361.mp4
